src/zkbob/ZkBobPool.sol

```diff
@@ -24,18 +24,19 @@ import "../utils/Ownable.sol";
 import "../proxy/EIP1967Admin.sol";
 import "../interfaces/IEnergyRedeemer.sol";
 import "../utils/ExternalSload.sol";
+import {Queue, PendingCommitment} from "./utils/Queue.sol";
 
 /**
  * @title ZkBobPool
  * Shielded transactions pool
  */
 abstract contract ZkBobPool is IZkBobPool, EIP1967Admin, Ownable, Parameters, ExternalSload {
     using SafeERC20 for IERC20;
+    using Queue for Queue.CommitmentQueue;
 
     uint256 internal constant MAX_POOL_ID = 0xffffff;
     bytes4 internal constant MESSAGE_PREFIX_COMMON_V1 = 0x00000000;
-    uint256 internal constant FORCED_EXIT_MIN_DELAY = 1 hours;
-    uint256 internal constant FORCED_EXIT_MAX_DELAY = 24 hours;
+    bytes4 internal constant MESSAGE_PREFIX_COMMON_V2 = 0x00000002;
```

Comment:
Should we rename this and other corresponding constants in a more clear way? e.g. `MESSAGE_ENCRYPTION_ECDH` (0), `MESSAGE_ENCRYPTION_PLAIN` (1), `MESSAGE_ENCRYPTION_SYMMETRIC` (2)

Reply:
Yeah, I've considered it. It appears to me that memo encryption falls outside the scope of contract logic. Additionally, `MESSAGE_PREFIX_COMMON_V1` could be removed once the system stabilizes. Therefore, from the contract perspective, there would be two possible prefixes: one for common transactions and one for dd transactions. That's why I prefer the current names. What do you think?

Reply:
Of course I understand your logic. I'm afraid the naming differences between the library and contract code can confuse thirdparty. It's unclear the prefix is the same as an encryption scheme. Btw the memo encryption scheme is really outside of contract logic. But the scheme type is under that scope since we analyze it correctness here.

But it's a minor thing anyway and I don't insist here :)

Reply:
Yes, it is a bit confusing indeed. I've tried to add some comments to clarify this, but all my attempts so far sound even more confusing than it currently is :) Let's keep it as is for now.